activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -17,6 +17,45 @@ def to_hash
         @config.merge(name: @name)
       end
 
+      class ConnectionConfigurations  #:nodoc:
+
+        def initialize(config)
+          @config = config
+          @root_level = nil
+          if url = ENV["DATABASE_URL"]
+            @database_url_config = ConnectionUrlResolver.new(url).to_hash
+          else
+            @database_url_config = nil
+          end
+        end
+
+        attr_accessor :root_level
+
+        def [](key)
+          ret = nil
+          if @root_level && c = @config[@root_level]
+            ret = c[key]
+          end
+          ret ||= @config[key]
+          if @database_url_config
+            ret = (ret || {}).merge(@database_url_config)
```

Comment:
Isn't this applying the URL-supplied database to every configuration we return?

That's wrong, and I'm alarmed the tests aren't point that out. It's hard to see what's going on with them, because they're all renamed at the same time, though. 😟